app/controllers/purchases_controller.rb

```diff
@@ -0,0 +1,46 @@
+class PurchasesController < ApplicationController
+  before_action :authenticate_user!, only: [:index]
+  before_action :set_item, only: [:index, :new, :create]
+
+  def index
+    gon.public_key = ENV["PAYJP_PUBLIC_KEY"]
+    @purchase_form = PurchaseForm.new
+    redirect_to root_path unless current_user.id != @item.user.id && @item.purchase.nil?
+  end
+
+  def new
+    @purchase_from = PurchaseForm.new
+  end
```

Comment:
```suggestion
  def new
    @purchase_from = PurchaseForm.new
  end
```
こちらのアクションは使用していないため削除しておきましょう。今回の購入ページはindexアクションを使用しています。